File: HW1/cs333_assignment1/KidneyDonation/python/donorcycle.py

def isInCycle(match_scores,donor_friends,query):
    return False

def take_input():
    n = int(input())
    m = int(input())
    donor_friends = input().split(',')
    for i in range(len(donor_friends)):
        donor_friends[i] = int(donor_friends[i])
    match_scores = []
    for i in range(m):
        match_row = input().split(',')
        for j in range(len(match_row)):
            match_row[j] = int(match_row[j])
        match_scores.append(match_row)
    query = int(input())
    print(isInCycle(match_scores,donor_friends,query))

take_input()

File: HW1/cs333_assignment1/KidneyDonation/java/DonorCycle.java
import java.io.*;
import java.util.*;

public class DonorCycle{

	public static boolean isInCycle(int[][] matchScores, int[] donorFriends, int query){
		return false;
	}

	public static void main(String[] args){

		Scanner s = new Scanner(System.in);

		int n = s.nextInt();
		int m = s.nextInt();

		String donorsLine = s.next();
		String[] donorsArray = donorsLine.split(",",0);
		int[] donorsFriends = new int[donorsArray.length];
		for(int i=0; i<m; i++){
			donorsFriends[i] = Integer.parseInt(donorsArray[i]);
		}

		int[][] matchScores = new int[m][n];
		for(int i=0; i<m; i++){
			String matchscoreLine = s.next();
			String[] matchscoreArray = matchscoreLine.split(",",0);
			for(int j=0; j<n; j++){
				matchScores[i][j] = Integer.parseInt(matchscoreArray[j]);
			}	
		}
		int query = s.nextInt();
		System.out.println(isInCycle(matchScores, donorsFriends, query));
	}


}
